### PR TITLE
NAS-124692 / 24.04 / Allow adding extra files to debug via a configuration option

### PR DESCRIPTION
**Pull request by themylogin**

### CODE REVIEW

**Comment by themylogin** [comment]
This is necessary for https://github.com/truenas/middleware/pull/12699

File: ixdiagnose/run.py [hunk 1]
```diff
@@ -34,6 +36,13 @@ def generate_debug() -> str:
     send_event(0, 'Generating debug')
     generate_plugins_debug(total_percentage=90)
     gather_artifacts(90, total_percentage=8)
+    for path, contents in conf.extra.items():
+        target_path = os.path.normpath(os.path.join(get_debug_dir(), path))
```
Comment (yocalebo) [inline ixdiagnose/run.py:40]: `get_debug_dir()` can be called once outside the outer for loop, right?

File: ixdiagnose/config.py [hunk 2]
```diff
@@ -25,6 +26,7 @@ def __init__(self):
         self.debug_path: Optional[str] = None
         self.exclude_artifacts: List[str] = []
         self.exclude_plugins: List[str] = []
+        self.extra: Dict[str, str] = {}
```
Comment (sonicaj) [inline ixdiagnose/config.py:29]: Should we rename this to `extra_artifacts` perhaps or something so it reflects what extra it is going to actually do ?

File: ixdiagnose/config.py [hunk 3]
```diff
@@ -13,6 +13,7 @@ class Configuration:
             'debug_path': {'type': ['string', 'null']},
             'exclude_artifacts': {'type': 'array', 'items': {'type': 'string'}},
             'exclude_plugins': {'type': 'array', 'items': {'type': 'string'}},
+            'extra': {'type': 'object'},
```
Comment (sonicaj) [inline ixdiagnose/config.py:16]: We should probably specify something like this as well so it validates that the values provided in this dict are actually string based and not integer/list or something else
```
    'type': 'object',
    'patternProperties': {
        '.*': {
            'type': 'array',
            'items': {'type': 'string'},
        }
    },
```

**Pull request closed without merging**